src/rollup/utils/persist/expiring-local-storage.js

```diff
@@ -0,0 +1,53 @@
+var PAYLOAD_KEY = '$';
+var EXPIRES_KEY = '@';
+
+function safeJsonParse (json) {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    // recover
+  }
+}
+
+export default {
+  getItem: function (key) {
+    var value = safeJsonParse(localStorage.getItem(key));
+
+    if (value && value[EXPIRES_KEY]) {
+      if (Date.parse(value[EXPIRES_KEY]) < Date.now()) {
```

Comment:
correct me if im wrong but it looks like this only gets ran when doing a read. which means the key (widget) is being evaluated by pathora, in which case we want to extend the expiration date. this looks like it will actually result in that stored value from being removed from localstorage.

e.g. i've closed a modal 30 days ago. i go back to the site 30days and 1 hour later but still initialie the widget. it triggers a read and as a result wipes that expired value when it should probably actually extend the expiration of that value. 

this likely brings up a couple questions / concerns with how we are handling expirations in general so it might be worth chatting and mapping that out so we are all on the same page and have something documented. 

might be the case that we need a flush method of sorts that looks at everything in local storage and expires them when they haven't been initialized. that flush method gets called on pathfora init. it would still result in the above example being wiped but it wouldn't be associated with the read....if that makes sense.

Reply:
one more thing we need to think about is how this reacts when the `priority` https://github.com/lytics/pathforajs/pull/299 option is passed. need to confirm this but i'd imagine in this case it stops evaluating the initialized widgets once one can be shown. this would also probably have the side effect of the expiration dates not being extended properly for the other widgets?

Reply:
defaulting to one year basically eliminates all of the above edge cases so maybe we focus on that for MVP and then regroup on how it should function in a perfect world?

Reply:
pretty sure this is not different from https://github.com/lytics/pathforajs/pull/350/files/f5300141314955d7db6fd95bae0724cb07e7b735#r338871395

We should be extending the expiration date on read

Reply:
Updated